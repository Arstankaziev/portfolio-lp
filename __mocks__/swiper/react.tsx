import React from 'react'

const SwiperSlide: React.FC = ({ children }) => <>{children}</>
const Swiper: React.FC = ({ children }) => <>{children}</>

export { SwiperSlide, Swiper }
