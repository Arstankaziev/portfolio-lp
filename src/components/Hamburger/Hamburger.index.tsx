import MenuToggle from './MenuToggle/MenuToggle.index'

interface HamburgerProps {
   handleOnClick: () => void
}

const Hamburger = ({ handleOnClick }: HamburgerProps) => (
   <MenuToggle handleOnClick={handleOnClick} />
)

export default Hamburger
