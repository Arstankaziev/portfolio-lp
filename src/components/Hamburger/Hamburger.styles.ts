import styled from 'styled-components'

export const Wrapper = styled.label`
   display: flex;
   flex-direction: column;
   width: 35px;
   cursor: pointer;
   gap: 7px;

   & span {
      background: black;
      border-radius: 10px;
      height: 4px;
      transition: 0.4s cubic-bezier(0.68, -0.6, 0.32, 1.6);
   }

   span:nth-of-type(1) {
      width: 50%;
   }

   span:nth-of-type(2) {
      width: 100%;
   }

   span:nth-of-type(3) {
      width: 75%;
   }
   & input {
      display: none;
   }

   & input:checked ~ span:nth-of-type(1) {
      transform-origin: bottom;
      transform: rotatez(45deg) translate(8px, 0px);
   }
   & input:checked ~ span:nth-of-type(2) {
      transform-origin: top;
      transform: rotatez(-45deg);
   }
   & input:checked ~ span:nth-of-type(3) {
      transform-origin: bottom;
      width: 50%;
      transform: translate(30px, -11px) rotatez(45deg);
   }
`
