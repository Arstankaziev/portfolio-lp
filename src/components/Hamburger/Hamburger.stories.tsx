import Hamburger from './Hamburger.index'
import { Meta, Story } from '@storybook/react'

export default {
   title: 'Hamburger',
   component: Hamburger
} as Meta

export const Basic: Story = (args) => <Hamburger {...args} />
