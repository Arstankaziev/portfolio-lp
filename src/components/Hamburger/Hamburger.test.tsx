import { render } from '@testing-library/react'
import Hamburger from './Hamburger.index'

describe('<Hamburger />', () => {
   it('should render the heading', () => {
      const layout = render(<Hamburger />)

      expect(layout).toMatchSnapshot()
   })
})
