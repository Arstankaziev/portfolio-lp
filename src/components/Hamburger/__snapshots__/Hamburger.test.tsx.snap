// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hamburger /> should render the heading 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  position: absolute;
  right: 1rem;
  top: 1rem;
  outline: none;
  border: none;
  cursor: pointer;
  width: 50px;
  height: 50px;
  border-radius: 50%;
  background: transparent;
}

@media (min-width:768px) {
  .c0 {
    right: 3rem;
    top: 2rem;
  }

  .c0 svg {
    width: 2.5rem;
    height: 2.5rem;
  }
}

<body>
    <div>
      <button
        class="c0"
      >
        <svg
          height="23"
          viewBox="0 0 23 23"
          width="23"
        >
          <path
            fill="transparent"
            stroke="white"
            stroke-linecap="square"
            stroke-width="2"
          />
          <path
            d="M 2 9.423 L 20 9.423"
            fill="transparent"
            stroke="white"
            stroke-linecap="square"
            stroke-width="2"
          />
          <path
            fill="transparent"
            stroke="white"
            stroke-linecap="square"
            stroke-width="2"
          />
        </svg>
      </button>
    </div>
  </body>,
  "container": .c0 {
  position: absolute;
  right: 1rem;
  top: 1rem;
  outline: none;
  border: none;
  cursor: pointer;
  width: 50px;
  height: 50px;
  border-radius: 50%;
  background: transparent;
}

@media (min-width:768px) {
  .c0 {
    right: 3rem;
    top: 2rem;
  }

  .c0 svg {
    width: 2.5rem;
    height: 2.5rem;
  }
}

<div>
    <button
      class="c0"
    >
      <svg
        height="23"
        viewBox="0 0 23 23"
        width="23"
      >
        <path
          fill="transparent"
          stroke="white"
          stroke-linecap="square"
          stroke-width="2"
        />
        <path
          d="M 2 9.423 L 20 9.423"
          fill="transparent"
          stroke="white"
          stroke-linecap="square"
          stroke-width="2"
        />
        <path
          fill="transparent"
          stroke="white"
          stroke-linecap="square"
          stroke-width="2"
        />
      </svg>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`
